src/ast/passes/codegen_llvm.cpp

```diff
@@ -3063,19 +3068,23 @@ std::tuple<Value *, CodegenLLVM::ScopedExprDeleter> CodegenLLVM::getMapKey(
     }
   } else {
     // No map key (e.g., @ = 1;). Use 0 as a key.
-    key = b_.CreateAllocaBPF(CreateUInt64(), map.ident + "_key");
+    assert(alloca_created_here);
+    key = b_.CreateMapKeyAllocation(CreateUInt64(),
+                                    map.ident + "_key",
+                                    map.loc);
     b_.CreateStore(b_.getInt64(0), key);
```

Comment:
This doesn't scale with the types used in the script, so perhaps better to keep this a stack allocation? That way we don't need unnecessary preallocation in scratch map.

Reply:
I thought about this too but I propose we keep it this way because this makes the definition of on_stack_limit consistent e.g. if I set on_stack_limit = 0, all allocations of map keys will be on the scratch buffer. This also makes testing easier as I can run standard runtime test cases with on_stack_limit = 0 instead of having to know that default maps don't use the scratch buffer.

The good news is in real usage, this will always be on the stack if on_stack_limit >= 8 - by default, its 32.

Reply:
Makes sense. Consistency and the fact that on_stack_limit makes it a stack allocation in practice are valid reasons

Reply:
> I'm also not quite sure what you mean by this.

Oh, I just meant that it's not script dependent. ie it'll always be a 64 bit allocation regardless of what type the user puts in the map.